.github/workflows/workflow.yml

```
@@ -110,11 +119,31 @@ jobs:
         with:
           name: basilisk_runtime.compact.compressed.wasm
           path: target/release/wbuild/basilisk-runtime/basilisk_runtime.compact.compressed.wasm
+  
+
+  upload-config-to-s3:
+    name: upload configuration to S3
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      AWS_S3_BUCKET: ${{ secrets.AWS_S3_CONFIG_BUCKET }}/testnet/
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_REGION: 'eu-west-1'
+      FILE: './basilisk-raw.json'
+    steps:
+      - uses: actions/download-artifact@v2
+        name: Download config file
+        with:
+          name: basilisk-raw
+          path: .
+      - name: S3 Upload basilisk-raw.json
+        uses: tpaschalis/s3-sync-action@master
 
   destroy-runner-instance:
     name: destroy runner
     runs-on: ubuntu-latest
-    needs: build
+    needs: [build, upload-config-to-s3, build-and-publish-docker-image]
```

you don't have to wait for `upload-config-to-s3` as they're not running on that runner

```suggestion
    needs: build-and-publish-docker-image
```

---

We do, because the file is uploaded from this machine, when it start destroying we cannot get the file from it anymore

---

thats not true, you are uploading it from github ubuntu, see https://github.com/galacticcouncil/Basilisk-node/pull/263/files/8ba96bc1635496a9240397f0b28524889d67b934#diff-126bf89616b7daa3d14ebc882ad18666aaf1c3dae888c4ba306a66ec80758bc1R126

---

the file is already uploaded to github artifacts even before build job finishes